qiskit/mapper/_layout.py

```diff
@@ -69,6 +75,19 @@ def __delitem__(self, key):
         dict.__delitem__(self, self[key])
         dict.__delitem__(self, key)
 
+    @staticmethod
+    def checktype(thing):
+        """Checks if thing is a valid type"""
+        if thing is None:
+            return
+        if isinstance(thing, int):
+            return
+        if isinstance(thing, tuple) and \
+                len(thing) == 2 and \
+                isinstance(thing[0], Register) and isinstance(thing[1], int):
+            return
+        raise LayoutError('The element %s should be a Register or an integer' % (thing,))
```

Comment:
According to what the method is doing, this error message should be updated:
```suggestion
        raise LayoutError('The element %s should be a (Register, integer) tuple or an integer' % (thing,))
```